components/brave_wallet/browser/blockchain_registry.cc

```diff
@@ -106,26 +106,60 @@ void BlockchainRegistry::GetAllTokens(const std::string& chain_id,
   std::move(callback).Run(std::move(tokens_copy));
 }
 
-void BlockchainRegistry::GetBuyTokens(const std::string& chain_id,
+void BlockchainRegistry::GetBuyTokens(mojom::OnRampProvider provider,
+                                      const std::string& chain_id,
                                       GetBuyTokensCallback callback) {
   std::vector<brave_wallet::mojom::BlockchainTokenPtr> blockchain_buy_tokens;
-  for (auto token : *kBuyTokens) {
-    auto blockchain_token = brave_wallet::mojom::BlockchainToken::New();
-    *blockchain_token = token;
-    blockchain_buy_tokens.push_back(std::move(blockchain_token));
+  const std::vector<mojom::BlockchainToken>* buy_tokens = nullptr;
+  if (provider == mojom::OnRampProvider::kWyre)
+    buy_tokens = &GetWyreBuyTokens();
+  else if (provider == mojom::OnRampProvider::kRamp)
+    buy_tokens = &GetRampBuyTokens();
+
+  if (buy_tokens == nullptr) {
+    std::move(callback).Run(std::move(blockchain_buy_tokens));
+    return;
+  }
+
+  for (const auto& token : *buy_tokens) {
+    if (token.chain_id != chain_id) {
+      continue;
+    }
+
+    blockchain_buy_tokens.push_back(
+        brave_wallet::mojom::BlockchainToken::New(token));
   }
   std::move(callback).Run(std::move(blockchain_buy_tokens));
 }
 
-void BlockchainRegistry::GetBuyUrl(const std::string& chain_id,
+void BlockchainRegistry::GetBuyUrl(mojom::OnRampProvider provider,
+                                   const std::string& chain_id,
                                    const std::string& address,
                                    const std::string& symbol,
                                    const std::string& amount,
                                    GetBuyUrlCallback callback) {
-  std::string url = base::StringPrintf(kBuyUrl, address.c_str(), symbol.c_str(),
-                                       amount.c_str(), kWyreID);
-
-  std::move(callback).Run(url);
+  std::string url;
+  if (provider == mojom::OnRampProvider::kWyre) {
+    if (chain_id != mojom::kMainnetChainId) {
+      std::move(callback).Run(url, "UNSUPPORTED_NETWORK");
+      return;
+    }
+
+    url = base::StringPrintf(kWyreBuyUrl, address.c_str(), symbol.c_str(),
+                             amount.c_str(), kWyreID);
+    std::move(callback).Run(std::move(url), absl::nullopt);
+  } else if (provider == mojom::OnRampProvider::kRamp) {
+    if (chain_id != mojom::kMainnetChainId) {
+      std::move(callback).Run(url, "UNSUPPORTED_NETWORK");
+      return;
+    }
```

Comment:
that check could be moved before `if (provider == mojom::OnRampProvider::kWyre) {` to avoid duplication in both `if` and `else`

Reply:
I assume we won't support buying on any other providers on a non mainnet chains

Reply:
I actually realised the `if (chain_id != mojom::kMainnetChainId)` check doesn't make sense anymore. First of all, the buy URL doesn't rely on the chain ID, and secondly, we support multiple EVM networks now with Ramp integration. So going to remove this condition altogether, since clients are already selectively activating the Buy tab based on the current network.

We'll keep the `chain_id` argument in the function call in case there's a future onramp integration where the buy URL actually relies on the network.